jaiarobotics/jaiabot
# bug/data-offload-command/JAIA-1259

## Summary
* The download command was being sent 2^n times for the nth bot in the download queue
* The function `downloadBotsInOrder` was being called after the removal of each bot from the queue

## Testing
* Download all (see video posted in Jira ticket: https://jaia-innovation.atlassian.net/jira/software/c/projects/JAIA/boards/15?selectedIssue=JAIA-1259) (note: apply the code changes shown in the code editor to cause the downloads to finish in sim)
* Download 2 bots individually and watch them pop from the queue, then press download all to verify individual downloads did not break the download-all functionality
